File: J9Ch/src/J_8_Data_Structure/Optional_6/_124_Longest_Consecutive_Sequence_Medium.cpp
/** 124 Longest Consecutive Sequence */

class Solution {
public:
  /**
   * @param nums: A list of integers
   * @return an integer
   */
  int longestConsecutive(vector<int> &num) {
    // write you code here
    unordered_set<int> map(num.begin(), num.end());
    int ans = 0;
    for (int i = 0; i < num.size(); i++) {
      if (map.find(num[i]) != map.end()) {
        map.erase(num[i]);
        int pre = num[i] - 1;
        int next = num[i] + 1;
        while (map.find(pre) != map.end()) {
          map.erase(pre);
          pre--;
        }
        while (map.find(next) != map.end()) {
          map.erase(next);
          next++;
        }
        ans = max(ans, next - pre - 1);
      }
    }
    return ans;
  }
};

File: JS9Ch/src/J_8_Data_Structure/Optional_6/python_cpp/_124_Longest_Consecutive_Sequence_Medium.py
# /** 124 Longest Consecutive Sequence


class Solution:
    # @param num, a list of integer
    # @return an integer
    def longestConsecutive(self, num):
        num.sort()
        l = num[0]
        ans = 1
        tmp = 1
        for n in num:
            if(n - l == 0):
                continue;
            elif(n - l == 1):
                tmp += 1
            else:
                if tmp > ans:
                    ans = tmp
                tmp = 1
            l = n
        if tmp > ans:
            ans = tmp
        return ans